src/kernel_density_estimation.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+/* @flow */
+
+var stddev = require('./sample_standard_deviation');
+var interquartileRange = require('./interquartile_range');
+
+var SQRT_2PI = Math.sqrt(2 * Math.PI);
+
+/**
+ * [Well-known kernels](https://en.wikipedia.org/wiki/Kernel_(statistics)#Kernel_functions_in_common_use)
+ * @private
+ */
+var kernels /*: {[string]: (number) => number} */ = {
+    /**
+     * The gaussian kernel.
+     * @private
+     */
+    gaussian: function (u) {
+        return Math.exp(-0.5 * u * u) / SQRT_2PI;
+    }
+};
+
+/**
+ * Well known bandwidth selection methods
+ * @private
+ */
+var bandwidthMethods /*: {[string]: (Array<number>) => number} */ = {
+    /**
+     * The ["normal reference distribution"
+     * rule-of-thumb](https://stat.ethz.ch/R-manual/R-devel/library/MASS/html/bandwidth.nrd.html),
+     * a commonly used version of [Silverman's
+     * rule-of-thumb](https://en.wikipedia.org/wiki/Kernel_density_estimation#A_rule-of-thumb_bandwidth_estimator).
+     * @private
+     */
+    nrd: function (x /*: Array<number> */) {
+        var h = stddev(x);
```

Comment:
From the wikipedia page about this, it seems that "H" is a smoothing parameter called the bandwidth. Can we call it "bandwidth" instead of H?

Reply:
Ah -- I think the entire quantity being returned by this function is the smoothing parameter / "bandwidth".  `h` is just an intermediate value in the computation.  If we wanted to give it a meaningful name rather than `h`, maybe... `spread`?

Reply:
Although, tbh I think calling it `spread` imbues it with more suggested meaning than it should have.

Reply:
I went with just `s`, so as not to confusingly collide with `h` on the wikipedia page, but also not promise any especially interpretable meaning for the temporary quantity.